test/zdtm/static/stopped03.c

```diff
@@ -0,0 +1,160 @@
+#include <signal.h>
+#include <sys/wait.h>
+#include <sys/mman.h>
+
+#include "zdtmtst.h"
+#include "lock.h"
+
+const char *test_doc = "Check, that stopped by SIGTSTP tasks are restored correctly";
+const char *test_author = "Yuriy Vasiliev <yuriy.vasiliev@openvz.org>";
+
+#define STOP_SIGNO SIGTSTP
+const char *stop_sigstr = "SIGTSTP";
+enum {
+	FUTEX_INITIALIZED = 0,
+	TEST_CRIU,
+	TEST_CHECK,
+	TEST_DONE,
+	TEST_EXIT,
+	TEST_EMERGENCY_ABORT,
+};
+
+struct shared {
+	futex_t fstate;
+	int status;
+	int code;
+} * sh;
+
+static int new_pgrp(void)
+{
+	siginfo_t infop;
+	int ret = 1;
+	pid_t pid;
+
+	/*
+	 * Set the PGID to avoid creating an orphaned process group,
+	 * which is not to be affected by terminal-generated stop signals.
+	 */
+	setpgid(0, 0);
+
+	pid = test_fork();
+	if (pid < 0)
+		goto err_cr;
+
+	if (pid == 0) {
+		/* wait for TEST_EXIT or TEST_EMERGENCY_ABORT*/
+		futex_wait_while_lt(&sh->fstate, TEST_EXIT);
+		exit(0);
+	}
+
+	if (kill(pid, STOP_SIGNO)) {
+		pr_perror("Unable to send %s", stop_sigstr);
+		goto err_cr;
```

Comment:
Hi @uravas does this case works well when using cgroup to freeze dumpee？

Criu uses `ptrace(PTRACE_SEIZE..);` in (`compel_interrupt_task`) [https://github.com/checkpoint-restore/criu/blob/criu-dev/criu/seize.c#L257] to seize dumpees after freezing them.  And in this case criu will hang in D state( discussed in https://lore.kernel.org/lkml/20151117193419.GA9993@redhat.com/t/)

Reply:
Hi. No, all "stopped*" zdtm tests are fail when I try to launch them with --freezecg.